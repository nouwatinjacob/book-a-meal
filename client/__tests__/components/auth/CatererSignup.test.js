import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  CatererSignup,
  mapStateToProps
} from '../../../src/components/auth/CatererSignup.jsx';

const props = {
  signupAction: () => Promise.resolve(),
  isLoading: true,
  errorResponse: {
    message: 'Error message'
  }
};

describe('CatererSignup component', () => {
  const setup = () => shallow(<CatererSignup {...props} />);

  it('should render CatererSignup component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'onFormSubmit'
  when signup button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    wrapper.setState({
      catererData: {
        businessName: 'Stephen shop',
        ownerName: 'Stephen Aribaba',
        businessAddress: '12, lokoja street',
        email: 'caterer1@gmail.com',
        password: 'password',
        password_confirmation: 'password',
        userType: 'caterer'
      }
    });

    const onFormSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'onFormSubmit'
    );
    wrapper.instance().onFormSubmit(event);
    expect(onFormSubmitSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'onInputChange'
  when input value is changed`, (done) => {
    const wrapper = setup();

    const event = {
      target: {
        name: {
          businessName: 'Stephen shop',
          ownerName: 'Stephen Aribaba',
          businessAddress: '12, lokoja street',
          email: 'caterer1@gmail.com',
          password: 'password',
          password_confirmation: 'password',
          userType: 'caterer'
        }
      },
      preventDefault: jest.fn()
    };

    const onInputChangeSpy = jest.spyOn(
      wrapper.instance(),
      'onInputChange'
    );
    wrapper.instance().onInputChange(event);
    expect(onInputChangeSpy).toHaveBeenCalled();

    done();
  });
});
