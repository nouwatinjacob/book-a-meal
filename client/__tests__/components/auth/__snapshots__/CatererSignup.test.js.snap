// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatererSignup component should render CatererSignup component correctly 1`] = `
<div
  className="container"
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <div
      className="login"
    >
      <div
        className="login-form"
      >
        <Loader
          color="#ff9600"
          height="50"
          margin="2px"
          type="Rings"
          visible={true}
          width="100"
        />
        <h3>
          Caterer Registration
        </h3>
        <form
          onSubmit={[Function]}
        >
          <input
            name="businessName"
            onChange={[Function]}
            placeholder="Business Name"
            type="text"
            value=""
          />
          <br />
          <input
            name="ownerName"
            onChange={[Function]}
            placeholder="Owner's Name"
            type="text"
            value=""
          />
          <br />
          <input
            name="businessAddress"
            onChange={[Function]}
            placeholder="Business Address"
            type="text"
            value=""
          />
          <br />
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
          <br />
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
          <br />
          <input
            name="password_confirmation"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
          <br />
          <button
            className="button warning"
          >
            Signup
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
